playbook/app/pb_kits/playbook/pb_currency/currency.rb

```diff
@@ -84,10 +87,17 @@ def whole_value
         amount.split(".").first.to_s
       end
 
+      def abbreviated_value
+        value = amount.split(".").first.split(",").join("").to_i
+        number_to_human(value, units: { thousand: "K", million: "M", billion: "B", trillion: "T" })
+      end
+
       def units_element
         _, decimal_part = amount.split(".")
         if unit.nil?
-          decimal_part.nil? ? ".00" : ".#{decimal_part}"
+          if abbreviate == false
+            decimal_part.nil? ? ".00" : ".#{decimal_part}"
+          end
```

Comment:
I'm wondering if we can optimize this and say `if unit.nil? && abbreviate == false` ?

Reply:
I'll try it out

Reply:
done